compose/src/main/java/com/maplibre/compose/camera/MapViewCamera.kt

```diff
@@ -1,140 +1,89 @@
+    val MOTION: CameraMotion = CameraMotion.Fly(1000)
   }
 }
 
-// TODO: Create Defaults so all inits use the same values.
-// TODO: Handle CameraPitch in/out
 @Parcelize
 data class MapViewCamera(
-    val state: CameraState = CameraState.Centered(latitude = 0.0, longitude = 0.0),
-    var zoom: Double = MapViewCameraDefaults.zoom,
-    val pitch: CameraPitch = MapViewCameraDefaults.pitch,
-    val direction: Double = MapViewCameraDefaults.direction
+    val state: CameraState =
+        CameraState.Centered(
+            latitude = 0.0,
+            longitude = 0.0,
+            zoom = MapViewCameraDefaults.ZOOM,
+            pitch = MapViewCameraDefaults.PITCH,
+            direction = MapViewCameraDefaults.DIRECTION),
+    val pitchRange: CameraPitchRange = MapViewCameraDefaults.PITCH_RANGE,
+    val padding: CameraPadding = MapViewCameraDefaults.PADDING,
+    val motion: CameraMotion = MapViewCameraDefaults.MOTION
     // TODO: Last change reason (See
     // https://github.com/stadiamaps/maplibre-swiftui-dsl-playground/blob/main/Sources/MapLibreSwiftUI/Models/MapCamera/MapViewCamera.swift#L4C1-L53C2)
 ) : Parcelable {
 
-  init {
-    if (zoom < MapViewCameraDefaults.minZoom) {
-      zoom = MapViewCameraDefaults.minZoom
-    } else if (zoom > MapViewCameraDefaults.maxZoom) {
-      zoom = MapViewCameraDefaults.maxZoom
-    }
-  }
-
-  override fun toString(): String {
-    return "MapViewCamera(state=$state, zoom=$zoom, pitch=$pitch, direction=$direction)"
-  }
-
-  override fun equals(other: Any?): Boolean {
-    if (this === other) return true
-    if (javaClass != other?.javaClass) return false
-
-    other as MapViewCamera
-
-    if (state != other.state) return false
-    if (zoom != other.zoom) return false
-    if (pitch != other.pitch) return false
-    if (direction != other.direction) return false
-    return true
-  }
-
-  override fun hashCode(): Int {
-    var result = state.hashCode()
-    result = 31 * result + zoom.hashCode()
-    result = 31 * result + pitch.hashCode()
-    result = 31 * result + direction.hashCode()
-    return result
-  }
-
   companion object {
     val Default = MapViewCamera()
 
     fun Centered(
         latitude: Double,
         longitude: Double,
-        zoom: Double = MapViewCameraDefaults.zoom,
-        pitch: CameraPitch = MapViewCameraDefaults.pitch,
-        direction: Double = MapViewCameraDefaults.direction
-    ) = MapViewCamera(CameraState.Centered(latitude, longitude), zoom, pitch, direction)
+        zoom: Double = MapViewCameraDefaults.ZOOM,
+        pitch: Double = MapViewCameraDefaults.PITCH,
+        direction: Double = MapViewCameraDefaults.DIRECTION,
+        pitchRange: CameraPitchRange = MapViewCameraDefaults.PITCH_RANGE,
+        padding: CameraPadding = MapViewCameraDefaults.PADDING,
+        motion: CameraMotion = MapViewCameraDefaults.MOTION
+    ) =
+        MapViewCamera(
+            CameraState.Centered(latitude, longitude, validZoom(zoom), pitch, direction),
```

Comment:
Let's `validZoom` and `validPitch` to the contractors themselves here or on CameraState.